components/sidebar/CourseView.tsx

```diff
@@ -61,9 +69,47 @@ const CourseView = ({ setSearchQuery }: { setSearchQuery: Function }) => {
     sections,
   })
   const { allowedToEdit } = useTimetable({ id: router.query.id as string })
+  const [siblingCourseId, setSiblingCourseId] = useState<string | null>(null)
 
   const course = courses[identifier]
 
+  const [checkCourseExists] = useLazyQuery(CHECK_COURSE_EXISTS, {
+    errorPolicy: "all",
+  })
+
+  useEffect(() => {
+    // Cannot put async callback to useEffect, so wrap
+    // Check for the sibling course if it exists, and if it does,
+    // add the ID of that course to the state. If not, it's null
+
+    // Set it to null initially, so that if a sibling course from another course existed,
+    // say CSC108F and CSC108S, and then the user clicks on CSC304F which has no S sibling,
+    // the sibling does not erroneously display CSC108S
+    setSiblingCourseId(null)
+
+    // Run the query
+    ;(async () => {
+      if (!course) return
+
+      const siblingCourseId = computeSiblingCourseId(course)
+      if (siblingCourseId === null) return
+
+      const result = await checkCourseExists({
+        variables: {
+          id: siblingCourseId,
+        },
+      })
+
+      if (result.error) {
+        // I don't know how to show errors in sqrl :/ so return for now
```

Comment:
I think this failing is not the end of the world for now.

Reply:
Ok.